src/charm.py

```diff
@@ -65,7 +65,7 @@ def __init__(self, *args: typing.Any) -> None:
         )
         self._observability = Observability(charm=self, charm_state=self._charm_state)
 
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.config_changed, self.on_config_changed)
```

Comment:
Why is this changed to a public method?

Reply:
Well, because it will be invoked in the `database.py` module, so I have to make it public.

Reply:
Could the database module not do a `charm.on.config_changed.emit()` (not sure if that's how it'd reference a charm object, but you get the idea) to avoid this?

Reply:
That's a great idea, updated, thanks!